src/GraphQL/Types/Collections/SchemaTypes.cs

```diff
@@ -428,7 +428,11 @@ private void AddType(IGraphType type, TypeCollectionContext context)
             {
                 foreach (var objectInterface in obj.Interfaces.List)
                 {
-                    if (AddTypeIfNotRegistered(objectInterface, context) is IInterfaceGraphType interfaceInstance)
+                    object typeOrError = RebuildType(objectInterface, false, context.ClrToGraphTypeMappings);
+                    if (typeOrError is string error)
+                        throw new InvalidOperationException($"The GraphQL implemented type for '{type.Name}' could not be derived implicitly. " + error);
+                    var objectInterface2 = (Type)typeOrError;
+                    if (AddTypeIfNotRegistered(objectInterface2, context) is IInterfaceGraphType interfaceInstance)
```

Comment:
This change is necessary to support `Interfaces.Add(typeof(GraphQLClrOutputTypeReference<MyInterfaceType>));`.  It just runs the type through the `RebuildType` code which runs it through the `IGraphTypeMappingProvider`.  Really more of a bug fix, I'd say.

Reply:
ok